operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -565,9 +566,12 @@ func (c *cesManagerIdentity) InsertCEPInCache(cep *cilium_v2.CoreCiliumEndpoint,
 					continue
 				}
 				defer ces.backendMutex.RUnlock()
+				defer c.identityLock.RUnlock()
```

Comment:
Can you document somewhere in this function that the ordering of the locking here must always be taken this way? This will help prevent future developers from choosing the wrong ordering and causing a deadlock.

Reply:
Done. Added above.